cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -1756,22 +1756,24 @@ endif(tmva)
 
 #---Check for PyROOT---------------------------------------------------------------------
 if(pyroot)
-  if(fail-on-missing AND (NOT PYTHONLIBS_FOUND AND NOT Python2_Interpreter_Development_FOUND AND NOT Python3_Interpreter_Development_FOUND))
-    message(FATAL_ERROR "PyROOT: Python development package not found and pyroot component required"
-                        " (python executable: ${PYTHON_EXECUTABLE})")
-  elseif(NOT PYTHONLIBS_FOUND AND NOT Python2_Interpreter_Development_FOUND AND NOT Python3_Interpreter_Development_FOUND)
-    message(STATUS "PyROOT: Python development package not found for python ${PYTHON_EXECUTABLE}. Switching off pyroot option")
-    set(pyroot OFF CACHE BOOL "Disabled because Python development package was not found" FORCE)
-  endif()
-  mark_as_advanced(FORCE pyroot2 pyroot3)
-  if(fail-on-missing AND pyroot2 AND NOT Python2_Interpreter_Development_FOUND)
-    message(FATAL_ERROR "PyROOT2: Python2 development package not found and pyroot2 component required"
-                        " (python2 executable: ${Python2_EXECUTABLE})")
+
+  if(NOT PYTHONLIBS_FOUND AND NOT Python3_Development_FOUND)
+    if(fail-on-missing)
```

Comment:
Also a comment for later: I know the `fail-on-missing` is a controversial topic, but I think disabling `pyroot` if Python is not found is a bit of an extreme measure of automatic feature detection. Users expect PyROOT to work, and only discovering after the build that it's not there because of automatic disabling can be frustrating.